api/client/stats_helpers.go

```diff
@@ -165,12 +169,21 @@ func (s *containerStats) Collect(cli client.APIClient, streamStats bool, waitFir
 }
 
 func (s *containerStats) Display(w io.Writer) error {
-	s.mu.RLock()
-	defer s.mu.RUnlock()
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	// NOTE: if you change this format, you must also change the err format below!
+	format := "%s\t%.2f%%\t%s / %s\t%.2f%%\t%s / %s\t%s / %s\t%d\n"
 	if s.err != nil {
-		return s.err
+		format = "%s\t%s\t%s / %s\t%s\t%s / %s\t%s / %s\t%s\n"
+		errStr := "--"
+		fmt.Fprintf(w, format,
+			s.Name, errStr, errStr, errStr, errStr, errStr, errStr, errStr, errStr, errStr,
+		)
+		err := s.err
+		s.err = nil
```

Comment:
Should we really be setting s.err to nil here? I would it think be better to leave it in the error state until we get a successful parse of data from the daemon.